keps/sig-storage/3314-csi-volume-snapshot-delta/README.md

```diff
@@ -0,0 +1,1484 @@
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+All unit tests will be included in the out-of-tree CSI repositories, with no
+impact on the test coverage of the core packages.
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+The integration tests will cover the lifecycle of the `VolumeSnapshotDelta` and
+`DriverDiscovery` resources. Test cases will be included to ensure that the
+`VolumeSnapshotDelta` resource works only with the `CREATE` operation. The
+typical CRUD operations of a CRD will work with the `DriverDiscovery` resource.
+
+The validation logic of the `VolumeSnapshotDelta` resource will also be covered,
+to ensure the aggregated API server serves the CBT request only if the required
+`VolumeSnapshot` and `DriverDiscovery` resources exist. If they don't, the
+request will fail.
+
+The integration tests setup will require the following fixtures:
+
+* The `VolumeSnapshot` [controller][16]
+* The CSI [host path driver][14]
+* Inject a mock handler in the `cbt-aggapi` aggregated API server to return mock
+CBT entries.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+The e2e tests will extend the integration tests to run on an actual Kubernetes
+cluster, set up with `kubetest` per the sig-testing [e2e tests
+documentation][15].
+
+A sample client will be used to create a `VolumeSnapshotDelta` resource to
+initiate the CBT request to the aggregated API server. The aggregated API server
+discovers the mock CSI driver using its `DriverDiscovery` resource. The CBT
+request is then forwarded to the mock CSI driver where the CSI GRPC invocation
+happens. The mock response payload is then returned to the sample client for
+verification.
+
+The e2e tests setup will require the following fixtures:
```

Comment:
e2e tests should be added to out-of-tree kubernetes-csi repo.

Reply:
Ack. Let's re-visit this once we decide on where the CBT code should live.